### ISSUE TITLE
When you start using x-ray detector, then the game crashes because of lack of sound

### ISSUE THREAD

**Issue by Inveriss** [ja2-stracciatella/ja2-stracciatella#1966]
[//]: # (Thank you for taking the time to report an issue or propose an enhancement. Questions are also welcome.)
[//]: # (If you have more than one item to report, please open separate issues for each one!)

[//]: # (This text will not be visible and you can freely delete it. Check the Preview tab for a cleaner look.)

### What is wrong

Hello

Everything with the game was just fine until I went to Balime to create x-ray detector. While in battle was using it, then after a while the game crashes. It was not crashing at the same time I press the button.

I checked the code and in the "Utils / Sound_Controll.cc" in line 307 there is indeed such an entry, but when I unpacked Sounds.slf I did not found such file.

```

	SOUNDSDIR "/explosion 12.wav",                              // 286
	SOUNDSDIR "/drink from canteen male.wav",                   // 287
	
        SOUNDSDIR "/x ray activated.wav",                           // 288**
	
        SOUNDSDIR "/catch object.wav",                              // 289
	SOUNDSDIR "/fence open.wav",                                // 290
```

I renamed with "REMOTE ACTIVATE.WAV" because it should be the most similar sound and it seems that the game is not crashing anymore.

Thanks

[//]: # (For bugs, please fill in this table [otherwise delete it]. The lines don't have to stay aligned.)

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
|  release 0.21.0 | JA2 Gold 1.12 ENG  |    NO    |    YES     |

```
2024-04-03T12:48:53.0848086Z [INFO] game\GameLoop.cc: Version Label: Stracciatella v0.21.0-git
2024-04-03T12:48:53.0848086Z [INFO] game\GameLoop.cc: Version #:     Build 04.12.02
2024-04-03T12:48:53.0848086Z [ERROR] sgp\SoundMan.cc: SoundLoadDisk Error for "sounds/x ray activated.wav": openGameResForReading: VfsFile_open "sounds/x ray activated.wav": entity not found
```

[//]: # (Please also include:)
[//]: # (- how to reproduce the bug and, if possible, a save game)
[//]: # (- the operating system)
[//]: # (- in case you know, for enhancements, if this feature was already added by some other source mod like 1.13)

[//]: # (When pasting backtraces or any other logs with hashed enumerations, please format it as a code block.)
[//]: # (You can do that by selecting it and clicking the "<>" icon or manually putting three backticks [```] at each end.)

**Comment by selaux** [comment]
Is this reproducible without renaming the sound (meaning it crashes everytime the device is used)?

That the sound is missing was already identified [here](https://github.com/ja2-stracciatella/ja2-stracciatella/issues/521#issuecomment-296242224). But this error log should have no effect, as this simply results in the related function call returning `NULL` and the sound not playing. For the X-Ray device this shouldnt matter as the return value is [not used](https://github.com/ja2-stracciatella/ja2-stracciatella/blob/138f210cdaa3dd0dbe36e1313a15be48bf5c04bb/src/game/Tactical/Handle_Items.cc#L743).

So I wonder if that is an issue with the sound part, or the actual effects of using the X-Ray device.

**Comment by Inveriss** [comment]
Yes, it crashes everytime when device is used. But as I mentioned this is not happening at once. It is somehow delayed. Sometimes after several turns or after the battle when I'm leaving the sector etc.

**Comment by lynxlynxlynx** [comment]
Sounds like memory corruption then, not really related to the sound, perhaps not even the detector. But that's hard for you to debug, since it requires something like valgrind or ASAN (manual building).

**Comment by Inveriss** [comment]
Unfortunately my coding experience is near to zero. I can only compile the code with simple static changes like in this issue. I'm making a small mod for Stracciatella and when I hit some issue, then I'm just sharing with you guys.
